src/Carbon/Traits/Timestamp.php

```diff
@@ -32,7 +32,7 @@ public static function createFromTimestamp(
     ): static {
         $date = static::createFromTimestampUTC($timestamp);
 
-        return $timezone === null ? $date : $date->setTimezone($timezone);
+        return $timezone === null ? $date->setTimezone(date_default_timezone_get()) : $date->setTimezone($timezone);
```

Comment:
This is a breaking change (basically reverted to Carbon 2 behavior), so it won't be merged in master/3.x branch anyway.

Reply:
BTW, it's the reason why the tests fail when changing this.

Reply:
Understood. Feel free to kill this.